pkg/apis/v1alpha1/condition.go

```diff
@@ -0,0 +1,22 @@
+package v1alpha1
+
+import corev1 "k8s.io/api/core/v1"
+
+type ConditionType string
+
+const (
+	Ready    ConditionType = "Ready"
+	NotReady ConditionType = "NotReady"
+)
+
+// Condition defines condition of custom resource.
+type Condition struct {
+	// Type defines condition type.
+	Type ConditionType `json:"type"`
+	// Status defines status of condition type, True or False.
+	Status corev1.ConditionStatus `json:"status"`
+	// Reason shows a brief reason of condition.
```

Comment:
Could we also add this field?

// Last time the condition transitioned from one status to another.
	// This should be when the underlying condition changed. If that is not known, then using the time when
	// the API field changed is acceptable.
	// +required
	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`

Reply:
Added, but a minor question, we have kubebuilder label `+required` which means it's a a required field while we also have `omitempty` in json tag which indicate that it's option. Do we have conflict here？

Reply:
sorry, I think we can set // +optional for it.